src/Diagnostics.RuntimeHost/Services/DevOpsClient/DevOpsClient.cs

```diff
@@ -173,27 +175,43 @@ public async Task<object> MakePullRequestAsync(string sourceBranch, string targe
 
             try
             {
-                dictionary[requestId + "--result"] = await gitClient.CreatePullRequestAsync(dictionary[requestId + "--pr"], _project, _repoID);
-            }
-            catch (Exception ex)
-            {
-                if (ex.Message.Contains("An active pull request for the source and target branch already exists"))
+                dictionary[requestId + "--prList"] = await GetPRListAsync(dictionary[requestId + "--source"], dictionary[requestId + "--target"], requestId);
```

Comment:
Quick question regarding this - This dictionary is maintained in-memory per process right? What if the request lands to create a PR lands in a different instance? How do we handle that scenario?

Reply:
do you mean if the request for PR and the request for Push land in different instances? each request gets passed all the data it needs from the ui. they dont communicate with each other